src/osd/ReplicatedPG.cc

```diff
@@ -8303,6 +8301,7 @@ void ReplicatedPG::eval_repop(RepGather *repop)
 	 repop->on_committed.erase(p++)) {
       (*p)();
     }
+    publish_stats_to_osd();
```

Comment:
I think this should go a bit further down, here https://github.com/ceph/ceph/pull/8039/files#diff-72747d40a424e7b5404366b557ff12a3R8357 next to calc_min_last_complete_ondisk().  We could also do it as an on_complete lambda, but I think it makes sense to do this unconditionally.